### PR TITLE
[Auth user Iter. 4.1] Only authorized users can modify the database

### PR DESCRIPTION
**Pull request by kyriakipapadimitriou**
Only authorized users (staff) can create, edit, and destroy users, orgs, apps, engagements, and iterations. There should be a check therefore before any action is implemented to verify the user has authorization to modify the database.
Tests written for the feature.

### CODE REVIEW

**Review by kimhoang73 — APPROVED** [review]
looks good! I like the extensive testing of all cases!!

File: features/auth_user.feature [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    | id | engagement_id | end_date   |
+    | 1  | 1             | 2017-04-14 |
+    | 2  | 1             | 2017-04-28 |
+  
+Scenario: Happy Path - Can create new app, edit and destroy existing app if I am staff
+  #Story ID: 153042639
+  Given I am logged in as "user1"
+  And I follow "Apps"
+  And I press "New App"
+  Then I should see "New App"
```
Comment (csungwon) [inline features/auth_user.feature:35]: maybe you want to change to `Then I should be on new apps page`. The reason is that when you are not allowed to do some actions, you will probably redirected back to where you were before committing that action. In our case, the unauthorized user will probably redirect to apps page and there exist "New App" button. You can also change the step to `Then I should see "New App" within ".page-header"` or something similar to limit the scope.

File: features/auth_user.feature [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+  And I press "Back"
+  And I press "Destroy" for "app1"
+  Then I should see "App was successfully destroyed."
+  
+Scenario: Sad Path - Cannot create new app, edit and destroy existing app if I am not staff
+  #Story ID: 153042639
+  Given I am logged in as "user2"
+  And I follow "Apps"
+  And I press "New App"
+  Then I should see "error"
```
Comment (csungwon) [inline features/auth_user.feature:48]: Maybe want to write the actual error message the user should see so that we can implement the feature accordingly?

**Review by csungwon — APPROVED** [review]
I like the scenarios. They are very descriptive and easy to follow. In the implementation, fix what I mentioned in the comments.

**Pull request merged**